src/scripts/main.js

```diff
@@ -1,3 +1,355 @@
+      const row = allCells.filter(cell => cell.cellIndex === i);
+      const rowNumbers = row.map(cell => cell.innerText);
+
+      arrForPush.push(...rowNumbers);
+    };
+  } else {
+    const allCells = [...findAllCells(document)];
+
+    allCells.map(cell => arrForPush.push(cell.innerText));
+  };
+};
+
+function arrIsChanged(arrBefore, arrAfter) {
+  for (let i = 0; i < arrAfter.length; i++) {
+    if (arrAfter[i] !== arrBefore[i]) {
+      createNumber();
+      break;
+    };
+  };
+};
+
+function insertTable(toTableType, fromTableType) {
+  if (toTableType === 'vetical') {
+    if (fromTableType) {
+      turnTheTable(tbodyVertical, 'column');
+      tbodyHorisontal.remove();
+      table.append(tbodyVertical);
+    };
+  }
+
+  if (toTableType === 'horisontal') {
+    if (!fromTableType) {
+      turnTheTable(tbodyHorisontal, 'row');
+      tbodyVertical.remove();
+      table.append(tbodyHorisontal);
+    };
+  }
+};
+
+function keydownListener(e) {
+  const cellsArr = [];
+
+  createArrWithAllNumbers(cellsArr);
+
+  const horisontalTable = !document.querySelector('tbody')
+    .classList.contains('tbody-columns');
+
+  if (e.key === 'ArrowUp') {
+    insertTable('vetical', horisontalTable);
+    rowsRewriteToLeft();
+  };
+
+  if (e.key === 'ArrowDown') {
+    insertTable('vetical', horisontalTable);
+    rowsRewriteToRight();
+  };
+
+  if (e.key === 'ArrowLeft') {
+    insertTable('horisontal', horisontalTable);
+    rowsRewriteToLeft();
+  };
+
+  if (e.key === 'ArrowRight') {
+    insertTable('horisontal', horisontalTable);
+    rowsRewriteToRight();
+  };
+
+  const newCellArr = [];
+
+  createArrWithAllNumbers(newCellArr);
+
+  arrIsChanged(cellsArr, newCellArr);
+
+  victory(newCellArr);
+
+  loss(newCellArr, horisontalTable);
+
+  addClass([...findAllCells(document)]);
+  scoreInGame.innerText = score;
+};
+
+function victory(valuesInCells) {
+  if (valuesInCells.find(cell => cell === '2048')) {
+    changeMessage('win');
+
+    document.removeEventListener('keydown', keydownListener);
+  };
+};
+
+function loss(valuesInCells, tableType) {
+  const emptyCells = valuesInCells.filter(cell => cell === '');
+
+  if (emptyCells.length === 0) {
+    const horisontalNembers = {};
+    const verticalNumbers = {};
+
+    const allCells = findAllCells(document);
+
+    if (!tableType) {
+      for (let i = 0; i < [...allCells].length; i++) {
+        if (!horisontalNembers[allCells[i].cellIndex]) {
+          horisontalNembers[allCells[i].cellIndex] = [];
+        }
+        horisontalNembers[allCells[i].cellIndex].push(allCells[i].innerText);
+
+        if (!verticalNumbers[allCells[i].parentElement.rowIndex]) {
+          verticalNumbers[allCells[i].parentElement.rowIndex] = [];
+        };
+
+        verticalNumbers[allCells[i].parentElement.rowIndex]
+          .push(allCells[i].innerText);
+      };
+    } else {
+      for (let i = 0; i < [...allCells].length; i++) {
```

Comment:
In both cases you use the same loop with similar logic. 
Try to simplify your code, and use only one loop